### PR TITLE
ROB: Fix errors/warnings on no /Resources within extract_text

### LINKED ISSUE #1272: KeyError: '/Resources'
See https://github.com/py-pdf/PyPDF2/issues/1269 for further details.

## Environment

Which environment were you using when you encountered the problem?

```bash
$ python -m platform
Linux-5.4.0-122-generic-x86_64-with-glibc2.29

$ python -c "import PyPDF2;print(PyPDF2.__version__)"
2.10.3
```
## Code + PDF

This is a minimal, complete example that shows the issue:

```python
from PyPDF2 import PdfReader
with open("TelemetryTX_EM.pdf", "rb") as f:
    reader = PdfReader(f, strict=False)
    full_content = " ".join([page.extract_text() for page in reader.pages])
```

PDF used above: [main.pdf](https://github.com/eagletrt/wiki/blob/0f3f16309604f665a47595c890d15af1b3aec6d6/fenice-telemetry-tx/PCB%20Outputs/Pdf/Edge%20Mount%20SMA/TelemetryTX_EM.pdf)

## Traceback

This is the complete Traceback I see:

```
Traceback (most recent call last):
  File "test4.py", line 4, in <module>
    content = " ".join([page.extract_text() for page in reader.pages])
  File "test4.py", line 4, in <listcomp>
    content = " ".join([page.extract_text() for page in reader.pages])
  File "/usr/local/lib/python3.8/dist-packages/PyPDF2/_page.py", line 1510, in extract_text
    return self._extract_text(
  File "/usr/local/lib/python3.8/dist-packages/PyPDF2/_page.py", line 1143, in _extract_text
    resources_dict = cast(DictionaryObject, obj["/Resources"])
  File "/usr/local/lib/python3.8/dist-packages/PyPDF2/generic/_data_structures.py", line 150, in __getitem__
    return dict.__getitem__(self, key).get_object()
KeyError: '/Resources'
```

### PR DESCRIPTION
**Pull request by pubpub-zz**
fix  #1272 (in text) and #1269 (in Xform)

### CODE REVIEW

File: tests/test_page.py [hunk 1]
```diff
@@ -238,6 +238,13 @@ def test_extract_text_single_quote_op():
         page.extract_text()
 
 
+def test_no_ressources_on_text_extract():
+    url = "https://raw.githubusercontent.com/eagletrt/wiki/0f3f16309604f665a47595c890d15af1b3aec6d6/fenice-telemetry-tx/PCB%20Outputs/Pdf/Edge%20Mount%20SMA/TelemetryTX_EM.pdf"
```
Comment (MasterOdin) [inline tests/test_page.py:242]: Better to not rely on a random third-party repo I think, and instead attach the PDF into the ticket and can link that:
```suggestion
    url = "https://github.com/py-pdf/PyPDF2/files/9428434/TelemetryTX_EM.pdf"
```

File: tests/test_page.py [hunk 2]
```diff
@@ -280,7 +287,7 @@ def test_extract_text_page_pdf_impossible_decode_xform(caplog):
     for page in reader.pages:
         page.extract_text()
     warn_msgs = normalize_warnings(caplog.text)
-    assert warn_msgs == [" impossible to decode XFormObject /Meta203"]
+    assert warn_msgs == [""]  # text extraction recognise no texg
```
Comment (MasterOdin) [inline tests/test_page.py:290]:
```suggestion
    assert warn_msgs == [""]  # text extraction recognise no text
```

File: tests/test_page.py (hunk 1 above)
Comment (pubpub-zz) [reply tests/test_page.py:242]: use of the referenced file from the issue. thanks for the recommendation

**Comment by MartinThoma** [comment]
Thank you for the PR @pubpub-zz :hugs:

**Comment by MartinThoma** [comment]
According to TABLE 3.27 Entries in a page object in the PDF 1.7 specs, the `/Resources` is required:

> (Required; **inheritable**) A dictionary containing any resources required by
the page (see Section 3.7.2, “Resource Dictionaries”). If the page requires
no resources, the value of this entry should be an empty dictionary. Omit-
ting the entry entirely indicates that the resources are to be inherited from
an ancestor node in the page tree.

As it is required, I would throw an exception in strict mode and a `logger_warning` otherwise.

Do you know if we handle the inheritance case? From what can it be inherited?

**Comment by pubpub-zz** [comment]
I've added some code to check in the parents. But if not present. I do not think it worth to check that Tj or TJ is present in the content (it will complexify the code with very limited interest)
if somebody exhibits a file showing this case, we can add the test at that time

**Pull request merged**